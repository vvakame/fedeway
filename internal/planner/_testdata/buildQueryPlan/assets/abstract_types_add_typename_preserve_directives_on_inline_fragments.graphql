# schema: supergraphSdl.graphqls

# for abstract types
# should preserve directives on inline fragments even if the fragment is otherwise useless

# TODO query ($b: Boolean!) { ← 変数定義部分がPlanに出力されるけどあっていいのかこれ？
query myQuery($b: Boolean!) {
    body {
        ... on Image {
            ... on NamedObject @include(if: $b) {
                name
            }
        }
    }
}
