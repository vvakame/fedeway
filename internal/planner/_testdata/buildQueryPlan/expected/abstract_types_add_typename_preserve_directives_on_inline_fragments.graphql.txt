QueryPlan {
	Fetch(service: "documents") {
		query ($b: Boolean!) {
			body {
				__typename
				... on Image {
					... on NamedObject @include(if: $b) {
						name
					}
				}
			}
		}
	},
}